guidelines/commit_and_pr_text.md

```diff
@@ -0,0 +1,55 @@
+
+# PR Metadata
+
+__PR title:__ Bug NNNN: A short sentence with capitalization and no period or the Jira tag for the issue “Feature TEAM-XXX:”
+
+* Try to describe the problem the user sees, rather than the code change you are doing
+  * I.e. “Console does not work when user clicks X” instead of “Fix drawer notification integration with underlying controller”
+* Prefix bug fixes with the bug number, “Bug XXXXX:” 
+  * (capitalize bug, space, number, then colon)
+* Do not let your PR title get truncated/wrapped into the description
+* Additional prefixes like what release the PR applies to are allowed, but they will be stripped from the changelog (the changelog for 4.4 doesn’t need to say “release-4.4” 
+  * Changelog strips anything between square brackets that has no spaces - i.e. [release-4.4] is stripped but [release 4.4] would not be
+    
+
+__PR description:__  Additional context reviewers need to understand your PR
+
+* Why was this change needed?
+* What changes are being made?  
+* Reference to any prior work(other PRs) or follow up work that will be needed.
+
+[Canonical example](https://github.com/openshift/openshift-apiserver/pull/69)
+
+# Git Commit Best practices
+
+## Number of Commits
+
+Strive for a single commit per PR.
+
+Minimal commits per PR make it easier to review the PR, easier to revert the PR if necessary, easier for someone doing “git blame” to follow the set of involved changes, and encourage smaller “chunks of work” (if it’s multiple commits, think about whether it should also be multiple PRs)
+
+Valid reasons to have multiple commits:
+
+* Separating implementation work from generated code or vendor updates
+* Separating api work from implementation
+* Moving files around in a refactor (do the file move in one commit, edit the files in the second commit, so that the edits stand out in the commit diff)
+* Temporarily adding additional commits in response to reviewer feedback so reviewers can easily see how their feedback was addressed.  These commits should be squashed prior to merging.
+
+Invalid reasons to have multiple commits
+* Separating tests+implementation
+* Committing+pushing your work each day/hour (use amend instead, or at least ensure you squash before merging).
+
+## Git Commit text
+
+__First line:__  An english language sentence with capitalization, no period, max 50 characters
+
+__Additional information:__  Why was this change needed?  What is the gist of the change?
+
+* Wrap lines at 72 characters, barring exceptions for links or other content that can't be wrapped.
+* Avoid referencing github issues/PRs if you are repeatedly amending the commit as it generates a lot of noise in the referenced issue/PR.
+
```

Comment:
I don't know if it's worth insisting that the "Additional information:" part is very important, and not optional? I've seen too many one line commit logs in openshift repositories which are quite cryptic for the occasional reader of the code :(

github issue references in logs are very useful when using `git blame` or similar. I've been told that instead of amending a commit/force pushing, one can use `--fixup` and that github will squash all the fixups at merge time regardless of the merge settings on the branch (I haven't tried it yet...)

Reply:
> I don't know if it's worth insisting that the "Additional information:" part is very important, and not optional? 

I can reword to make it a stronger request/suggestion, but I'd like to avoid absolute requirements/rules since:

1) right now we have no plans to enforce anyway
2) "there's always an exception"

Reply:
Yup, of course there are always exceptions :) `Fix 'requierd' typo in doc` does not really more than than :)